File: electivehelper-api/src/main/java/cl/ufro/dci/electivehelper_api/domain/Slot.java
package cl.ufro.dci.electivehelper_api.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

@Data
@AllArgsConstructor
@ToString
public class Slot {
    private String period;
    private String day;
    private String startTime;
    private String endTime;
    private String place;
}

File: electivehelper-ui/src/app/domain/slot.ts
export class Slot{
    period: string;
    day: string;
    startTime: string;
    endTime: string;

    constructor(period: string, day: string, startTime: string, endTime: string){
        this.period = period;
        this.day = day;
        this.startTime = startTime;
        this.endTime = endTime;
    }

    equals(slot: Slot): boolean{
        return this.day === slot.day && this.period === slot.period;
    }
}